src/Legacy/Storage.php

```diff
@@ -2314,29 +2314,34 @@ private function parseWhereParameter($key, $value, $fieldtype = false)
      */
     public function getContentType($contenttypeslug)
     {
-        $contenttypeslug = $this->app['slugify']->slugify($contenttypeslug);
-
         // Return false if empty, can't find it.
         if (empty($contenttypeslug)) {
             return false;
         }
 
+        $contenttype = $this->app['config']->get('contenttypes/' . $contenttypeslug);
+
         // See if we've either given the correct contenttype, or try to find it by name or singular_name.
-        if ($this->app['config']->get('contenttypes/' . $contenttypeslug)) {
-            $contenttype = $this->app['config']->get('contenttypes/' . $contenttypeslug);
-        } else {
-            foreach ($this->app['config']->get('contenttypes') as $key => $ct) {
-                if (isset($ct['slug']) && ($contenttypeslug === $ct['slug'])) {
-                    $contenttype = $ct;
-                    break;
-                }
-                if (isset($ct['singular_slug']) && ($contenttypeslug === $ct['singular_slug'])) {
-                    $contenttype = $ct;
-                    break;
-                }
-                if ($contenttypeslug === $this->app['slugify']->slugify($ct['singular_name']) || $contenttypeslug == $this->app['slugify']->slugify($ct['name'])) {
-                    $contenttype = $ct;
-                    break;
+        if (!$contenttype) {
+            // Also check for the slugified version of the content type
+            $slugifiedContentType = $this->app['slugify']->slugify($contenttypeslug);
+            $contenttype = $this->app['config']->get('contenttypes/' . $slugifiedContentType);
+
+            if (!$contenttype) {
+                foreach ($this->app['config']->get('contenttypes') as $key => $ct) {
+                    if (
+                        (isset($ct['slug']) &&
+                            (($contenttypeslug === $ct['slug']) || ($slugifiedContentType === $ct['slug']))
+                        ) ||
+                        (isset($ct['singular_slug']) &&
+                            (($contenttypeslug === $ct['singular_slug']) || ($slugifiedContentType === $ct['singular_slug']))
+                        ) ||
+                        $slugifiedContentType === $this->app['slugify']->slugify($ct['name']) ||
+                        $slugifiedContentType === $this->app['slugify']->slugify($ct['singular_name'])
+                    ) {
+                        $contenttype = $ct;
+                        break;
+                    }
                 }
```

Comment:
So… this *is* legacy, but I can't help but think given everything we're dealing with here it would be easier to:

```
    public function getContentType($contentTypeSlug)
    {
        // Return false if empty, can't find it.
        if (empty($contentTypeSlug)) {
            return false;
        }

        $contentType = $this->app['config']->get('contenttypes/' . $contentTypeSlug);

        // See if we've either given the correct contenttype, or try to find it by name or singular_name.
        if (!$contentType) {
            // Also check for the slugified version of the content type
            $slugifiedContentType = $this->app['slugify']->slugify($contentTypeSlug);
            $contentType = $this->app['config']->get('contenttypes/' . $slugifiedContentType);

            if (!$contentType) {
                foreach ($this->app['config']->get('contenttypes') as $key => $ct) {
                    $isCtPlural = isset($ct['slug']) && ($contentTypeSlug === $ct['slug']) || ($slugifiedContentType === $ct['slug']);
                    $isCtSingle = isset($ct['singular_slug']) && ($contentTypeSlug === $ct['singular_slug']) || ($slugifiedContentType === $ct['singular_slug']);
                    $isCtPluralMod = $slugifiedContentType === $this->app['slugify']->slugify($ct['name']);
                    $isCtSingleMod = $slugifiedContentType === $this->app['slugify']->slugify($ct['singular_name']);

                    if ($isCtPlural || $isCtSingle || $isCtPluralMod || $isCtSingleMod) {
                        $contentType = $ct;
                        break;
                    }
                }
            }
        }

        if (!empty($contentType)) {
            return $contentType;
        } else {
            return false;
        }
    }
```

On one hand we lose OoE on the condition (and it is covering edge-case), but on the other it becomes a lot more readable.